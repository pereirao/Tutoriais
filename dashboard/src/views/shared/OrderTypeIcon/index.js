//@flow

import React from 'react';
import I18n from '../../../localization';

import shipped from './assets/box.png';
import pickup from './assets/bag.png';
import on_demand from './assets/truck.png';
import scheduled from './assets/clock.png';

const icon_map = { shipped, pickup, on_demand, scheduled };

type OrderTypeIconProps = {type: string};
const OrderTypeIcon = ({type}: OrderTypeIconProps) => (
  <img
    src={icon_map[type]}
    alt={I18n.t(`accessibility.icon.${type}`)} />
);

export default OrderTypeIcon;
