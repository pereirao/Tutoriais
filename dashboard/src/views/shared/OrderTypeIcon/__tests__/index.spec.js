import React from 'react';

import OrderTypeIcon from '../index';

it('renders correctly for scheduled type', () => {
  expect(shallow(<OrderTypeIcon type="scheduled" />)).toMatchSnapshot();
});
it('renders correctly for shipped type', () => {
  expect(shallow(<OrderTypeIcon type="shipped" />)).toMatchSnapshot();
});
it('renders correctly for pickup type', () => {
  expect(shallow(<OrderTypeIcon type="pickup" />)).toMatchSnapshot();
});
it('renders correctly for on_demand type', () => {
  expect(shallow(<OrderTypeIcon type="on_demand" />)).toMatchSnapshot();
});
