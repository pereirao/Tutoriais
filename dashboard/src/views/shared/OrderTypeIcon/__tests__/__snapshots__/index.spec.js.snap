// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for on_demand type 1`] = `
<img
  alt="Delivery Order"
  src="truck.png"
/>
`;

exports[`renders correctly for pickup type 1`] = `
<img
  alt="Pickup Order"
  src="bag.png"
/>
`;

exports[`renders correctly for scheduled type 1`] = `
<img
  alt="Scheduled Order"
  src="clock.png"
/>
`;

exports[`renders correctly for shipped type 1`] = `
<img
  alt="Shipping Order"
  src="box.png"
/>
`;
